src/pkg/egress/syslog/https.go

```diff
@@ -32,42 +44,103 @@ func NewHTTPSWriter(
 ) egress.WriteCloser {
 
 	client := httpClient(netConf, tlsConf)
-
-	return &HTTPSWriter{
-		url:             binding.URL,
-		appID:           binding.AppID,
-		hostname:        binding.Hostname,
-		client:          client,
-		egressMetric:    egressMetric,
-		syslogConverter: c,
+	if binding.URL.Query().Get("batching") == "true" {
+		return &BatchHTTPSWriter{
+			HTTPSWriter: HTTPSWriter{
+				url:             binding.URL,
+				appID:           binding.AppID,
+				hostname:        binding.Hostname,
+				client:          client,
+				egressMetric:    egressMetric,
+				syslogConverter: c,
+			},
+			batchSize:    BATCHSIZE,
+			sendInterval: time.Second,
+			egrMsgCount:  0,
+		}
+	} else {
+		return &HTTPSWriter{
+			url:             binding.URL,
+			appID:           binding.AppID,
+			hostname:        binding.Hostname,
+			client:          client,
+			egressMetric:    egressMetric,
+			syslogConverter: c,
+		}
 	}
 }
 
-func (w *HTTPSWriter) Write(env *loggregator_v2.Envelope) error {
+func (w *BatchHTTPSWriter) sendMsgBatch() error {
+	currentEgrCount := w.egrMsgCount
+	currentMsg := w.msgBatch.Bytes()
```

Comment:
Is `BatchHTTPSWriter.Write` meant to be thread-safe? If it is, then it seems like there's a possible race condition here: accessing `msgBatch` and `egrMsgCount` when another call to `BatchHTTPSWriter.Write` might be accessing them as well.

Reply:
I am not the most go native writer to be honest, if there is a way to make this thread safe with go,
If you could advise on how to do that, that would be appreciated.

Reply:
This is likely not meant to be threadsafe, I think we're okay.